### PR TITLE
RD-3100 Replaced error message popup with a modal

### PR DESCRIPTION
**Pull request by kubama**
## Description
Replaced error message popup with a modal. Reused and extended existing error causes modal as well as modal opening icon. The color of the icon was changed from red to black, as it may now be used in the context of non-error messages as well.

## Screenshots / Videos
image

image

image

## Checklist

- [X] My code follows the UI Code Style.
- [x] I verified that all tests and checks have passed.
- [X] I verified if that change requires a documentation update.
- [X] I added proper labels to this PR.

## Tests
https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/1266/

## Documentation
I will update widget documentation once this PR is approved.

### CODE REVIEW

**Review by qooban — COMMENTED** [review]
Looks good. I've got only one comment.

**Comment by qooban** [comment]
Tested UI and it works very well 👍🏻 

The only thing I noticed that could be improved is `HightlightText` component language setting. It should display preformatted text without strange coloring. We use `language="json"` which is not correct IMHO.

For example in this case:
Screenshot from 2021-09-15 10-09-02
we see that some numbers are colored and some not.

I found at https://github.com/react-syntax-highlighter/react-syntax-highlighter#props that for this case (just render plain monospaced text) we can use `text` language. It would require a change in the `cloudify-ui-components`, so I think it is out of the scope, but what do you think about registering a ticket for that?

File: widgets/events/src/EventsTable.tsx [hunk 1]
```diff
@@ -15,8 +15,7 @@ export default class EventsTable extends React.Component {
 
         this.state = {
             error: null,
-            errorCauses: [],
```
[the remark this answers is not in the document]
Comment (kubama) [reply widgets/events/src/EventsTable.tsx:18]: We don't need it, I removed it from `hideDetailsModal`.

**Comment by kubama** [comment]
> The only thing I noticed that could be improved is `HightlightText` component language setting. It should display preformatted text without strange coloring. We use `language="json"` which is not correct IMHO.

Yes, I agree. The simplest solution that could be applied right away is to remove `HightlightText` from message and traceback (which uses `python` as `language`, but it also leads to strange coloring) and replace it with a regular `code` tag (which is essentially what would `<HightlightText language="text">` do). What do you think?

**Comment by qooban** [comment]
> Yes, I agree. The simplest solution that could be applied right away is to remove `HightlightText` from message and traceback (which uses `python` as `language`, but it also leads to strange coloring) and replace it with a regular `code` tag (which is essentially what would `<HightlightText language="text">` do). What do you think?

Yes, agree, sounds like a good idea 👍🏻

**Review by qooban — APPROVED** [review]

**Review by qooban — APPROVED** [review]

**Review by stepek — APPROVED** [review]

**Pull request merged**